### DISCUSSION TITLE
Does loading UI works with generateStaticParams?

### DISCUSSION THREAD

**Discussion by hkmute** [vercel/next.js#48329] [Help]
### Summary

Suppose I have a page that uses generateStaticParams, when I navigate to the page which was not generated at build time, it seems that the loading UI is not shown and my UI freezes until the page is generated at server.

Is it possible to show the loading UI while generating the page?

**Comment by SamKomesarook** [comment]
Do the client components make any async calls that would slow it down? If so, consider wrapping that client component in a [suspense boundary](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming).
